File: backend/Bingol-University-Social-Media-main/src/main/java/bingol/campus/security/dto/LoginRequestDTO.java
package bingol.campus.security.dto;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class LoginRequestDTO {
    private String username; // Kullanıcı numarası
    private String password;   // Kullanıcı şifresi
    private String ipAddress;  // Kullanıcının giriş yaptığı IP adresi
    private String deviceInfo; // Kullanıcının giriş yaptığı cihaz bilgisi
}

File: frontend/lib/models/login_request_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'login_request_dto.g.dart';

@JsonSerializable()
class LoginRequestDTO {
  final String username; // Kullanıcı numarası
  final String password; // Kullanıcı şifresi
  final String ipAddress; // Kullanıcının giriş yaptığı IP adresi
  final String deviceInfo; // Kullanıcının giriş yaptığı cihaz bilgisi

  LoginRequestDTO({
    required this.username,
    required this.password,
    required this.ipAddress,
    required this.deviceInfo,
  });

  factory LoginRequestDTO.fromJson(Map<String, dynamic> json) => _$LoginRequestDTOFromJson(json);
  Map<String, dynamic> toJson() => _$LoginRequestDTOToJson(this);
} 